getlantern/fronted
# Adding SNI config

This pull request implements the SNI aribitrary configuration which allows flashlight to send a SNI on requests depending on the provider SNI configuration (and country at flashlight). Currently our requests doesn't send a SNI domain when making TLS connections (if you open wireshark `sudo wireshark -i <interface> -k `, scan and filter for `ClientHello`s - `_ws.col.info contains "Client Hello"` - , you should be able to see our requests without specifying the SNI domain. e.g. image with empty client hello below. 
![e.g. image with empty client hello below](https://github.com/user-attachments/assets/21f9c5ba-77a3-4834-8e48-f7d06579283e)

Here's a example of how it  should look like now when sending requests with the SNI config.
![image](https://github.com/user-attachments/assets/d669d387-ec0d-410d-90fd-a5531736834a)

## Testing
1. Run wireshark for capturing your local traffic:
```bash
$ sudo wireshark -i <interface> -k
```
2. At the wireshark interface filter the content by the ClientHello info, you can use the following filter:
```
_ws.col.info contains "Client Hello"
```
3. Execute the test:
```bash
$ go test -run=TestDirectDomainFrontingWithSNIConfig ./...
```
4. You should be able to see requests with SNIs at wireshark (with random domains)

> [!NOTE]
> This pull request is required to be merged before https://github.com/getlantern/flashlight/pull/1396
